UnderThere/Program.cs

```diff
@@ -332,35 +332,30 @@ public static void createItems(List<UTitem> Items, ILinkCache lk)
             List<UTitem> toRemove = new List<UTitem>();
             foreach (var item in Items)
             {
-                if (item.formID != null && item.formID.Length == 8)
-                {
-                    item.formID = item.formID.Substring(2, 6);
-                }
-                string itemKey = item.formID + ":" + item.source;
-                if (FormKey.TryFactory(itemKey, out var formKey) && formKey.IsNull == false)
+                if (FormKey.TryFactory(item.FormKey, out var formKey) && !formKey.IsNull)
                 {
                     // If conversion successful
                     if (!lk.TryResolve<IArmorGetter>(formKey, out var origItem))
                     {
-                        Console.WriteLine("Could not find item with formKey " + itemKey);
+                        Console.WriteLine("Could not find item with formKey " + formKey);
                         toRemove.Add(item);
                     }
                     else
                     {
-                        item.FormKey = formKey;
+                        item.FormKeyObject = formKey;
                         Armor ModdedItem = (Armor)origItem.DeepCopy();
                         ModdedItem.Name = item.Name;
                         item.FormLink = ModdedItem;
```

Comment:
I didn't adjust this in the PR, but I wanted to bring up this code section, as I don't think it's doing what you think it is.

Right now the logic is doing something like:
- Starting with some FormKey given to us by the input config file (let's say `000820:underwearforeveryone.esp`)
- Looking up the actual read-only Armor record object from disk for that FormKey
- Making a copy of that armor to a new settable object
- Assigning the `UTItem`'s FormLink to that new armor.  But this is the confusion, I think.  This will just grab the FormKey from that copied armor, nothing more;  It would be equivalent of `item.FormLink.FormKey = ModdedItem.FormKey` or even just `item.FormLink.FormKey = formKey`.  Notice the actual object `ModdedItem` is not really referenced.. it's just grabbing the FormKey.  Additionally, all the lookups/copies aren't adding anything, as we already had the FormKey right off the bat.  
- The copied settable armor is then lost to the void and gets cleaned up by the Garbage Collection, as no one is referencing it anymore.

It sounds like you might be trying to set up ahead of time an `Armor` object you'll eventually modify in some other code section?   If this is the case, you'll need to add it to the patch mod object for it to be retained and exported, and/or just use the GetOrAddAsOverride API:
`Armor armor = outgoingPatch.Armors.GetOrAddAsOverride(gettableArmor);`
This converts a readonly `IArmorGetter` to an `Armor` you can modify and has been added to the patch (or the existing one is given back to you, if it was already added to the outgoing patch)

The usage of FormLink as a member of UTItem I don't think is too useful.  It's mainly just an object used by Mutagen to limit what a FormKey can match against within a LinkCache.  (eg, a FormKey that matched to a Potion record, but was in a FormLink<Armor> would fail to match, as a Potion is not an Armor).  For you, though, you already have more interesting objects:  the already looked up `IArmorGetter` or even a `Armor` object added to the patch, potentially.